openquake/hazardlib/gsim/base.py

```diff
@@ -976,3 +978,180 @@ def __getitem__(self, imt):
             (co, (min_above[co] - max_below[co]) * ratio + max_below[co])
             for co in max_below.keys()
         )
+
+
+class GMPETable(GMPE):
+    """
```

Comment:
There is a serious problem with the approach used here in a cluster situation. The .hdf5 files are generally not visible to the workers (they are visible only if they live in a shared directory) so the GMPE cannot read the data. However it looks like the tables are small (~50k): that means that you can read the whole hdf5 file at instantiation time in the controller node, save it as an attribute of the GMPE instance (for instance as a dictionary of arrays), and then send the GMPE instance to the workers.

Also, do you really want to define so many classes? It looks like we could define a single class GSCCanada2015 and have different instances reading from different tables. For the user the difference would be that instead of writing in the GMPE logic tree file something like

`<uncertaintyModel>GSCCanada2015WOffshoreMed</uncertaintyModel>`

she would write

`<uncertaintyModel>GSCCanada2015("Woffshore_rhypo_med")</uncertaintyModel>`

Let's discuss our options on Monday.